### PR TITLE
util: move str_verscmp() and add tests

### PR DESCRIPTION
**Pull request by yuwata**

### CODE REVIEW

File: src/test/test-string-util.c [hunk 1]
```diff
@@ -886,6 +886,50 @@ static void test_string_contains_word(void) {
...
+        test_str_verscmp_one("124.457-10.1", "123.456-78.9", true);
+        test_str_verscmp_one("123.456-78.9", "123.456-78", false);
+        test_str_verscmp_one("123.456-78.9", "123.456-7", false);
+        test_str_verscmp_one("123.456-78.9", "123.456", false);
+        test_str_verscmp_one("123.456-78.9", "123.45", false);
+        test_str_verscmp_one("123.456-78.9", "123.4", false);
+        test_str_verscmp_one("123.456-78.9", "123", false);
+        test_str_verscmp_one("123.456-78.9", "12", false);
+        test_str_verscmp_one("123.456-78.9", "1", false);
+        test_str_verscmp_one("123~rc1", "123", false); /* Hmm?? */
```
Comment (yuwata) [inline src/test/test-string-util.c:929]: Should we handle this gracefully?

File: src/basic/string-util.c [hunk 2]
```diff
@@ -1133,3 +1133,66 @@ int string_contains_word_strv(const char *string, const char *separators, char *
                 *ret_word = found;
         return !!found;
 }
+
+static bool is_digit(int c) {
+        return c >= '0' && c <= '9';
+}
```
Comment (dnicolodi) [inline src/basic/string-util.c:1139]: Isn't this the same of `isdigit()` form `ctype.h` ?

File: src/test/test-string-util.c (hunk 1 above)
Comment (bluca) [reply src/test/test-string-util.c:929]: yeah, result should be the opposite

File: src/basic/string-util.c [hunk 3]
```diff
@@ -1133,3 +1133,66 @@ int string_contains_word_strv(const char *string, const char *separators, char *
...
+        if (c == 0 || is_digit(c))
+                return 0;
+
+        if ((c >= 'a') && (c <= 'z'))
+                return c;
+
+        return c + 0x10000;
+}
+
+int str_verscmp(const char *s1, const char *s2) {
```
Comment (poettering) [inline src/basic/string-util.c:1151]: we should probably find a different name for the function, it's just confusing that there's strverscmp() and str_verscmp() and they do the same thing but also quite differently.

maybe rename this strverscmp_improved() or strverscmp_extra() or strverscmp_with_a_twist() or so.. ;-)

**Comment by yuwata** [comment]
Replaced by #18416.

**Pull request closed without merging**